### ISSUE TITLE
Setup CI

### ISSUE THREAD

**Issue by poszu** [athenavm/athena-project-template#3]
To avoid regressions, the CI should at the very least:
- lint with `clippy`
- run the `script` (which will build and execute both `program` and `script`

**Comment by lrettig** [comment]
Makes sense. For the record note that this is already being done in the main repo:

https://github.com/athenavm/athena/blob/abb1d9765c8995601527fddd67aaa121562b8e9c/.github/workflows/pr.yml#L128-L167
